src/main/java/net/dv8tion/jda/api/interactions/components/text/Modal.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.interactions.components.text;
+
+import net.dv8tion.jda.api.interactions.components.ActionComponent;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.internal.interactions.component.ModalImpl;
+import net.dv8tion.jda.internal.utils.Checks;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a Discord Modal
+ *
+ * <p>Replying to an interaction with a modal will cause a form window to pop up on the User's client.
+ *
+ * <p><b>Only a maximum of 5 components can be included in a Modal, and only {@link net.dv8tion.jda.api.interactions.components.text.TextInput TextInputs} are allowed.</b>
+ *
+ * @see net.dv8tion.jda.api.events.interaction.ModalSubmitInteractionEvent
+ */
+public interface Modal extends ActionComponent
```

Comment:
Add a usage example

Reply:
Why is this an ActionComponent?

Reply:
Because you told me to
![image](https://user-images.githubusercontent.com/68189302/153643711-b3fae444-5792-49e8-8acb-751af9543117.png)

Reply:
oh nevermind thats for TextInputs